lib/iris/_lazy_data.py

```diff
@@ -93,6 +93,8 @@ def as_lazy_data(data, chunks=_MAX_CHUNK_SIZE):
         The input array converted to a dask array.
 
     """
+    if isinstance(data, ma.core.MaskedConstant):
+        data = ma.masked_array([data], mask=data.mask)
```

Comment:
Ah! You're right! That does indeed work.

I have now put up a new commit with that change in